utils/text_extractor.py

```diff
@@ -47,11 +53,22 @@ def get_all_text_from_url(url, retries=5):
         allText = h.handle(html)
 
         return allText, title
+    except timeout as exc:
+        logging.error(
+            "an exception occurred while trying to access url {} trying again \n more details: {}".format(url, exc))
```

Comment:
did you test it? after  "except" there should be a real Class that extend Exception class

Reply:
yes it is a real class of _socket that is the exception urllib raise on timeout